halo2_proofs/src/poly/commitment.rs

```diff
@@ -187,6 +193,116 @@ pub trait Verifier<'params, Scheme: CommitmentScheme> {
             > + Clone;
 }
 
+fn friendly_create_proof<
```

Comment:
It is the create proof function, but the prover only has to input the polynomials, the evaluation points and the commitment to the function, the rest of the work like creating the transcript and queries will be done by the function itself, the prover does not have to initialize the whole transcript and queries. In addition, we can use this function outside Halo2 without having to public the fields of ProverQuery, since the query will be created within this function in the crate, while the original  create_proof function requires the prover to initialize the query, which cannot be done outside the Halo2 crate.